wormhole-connect/src/utils/index.ts

```diff
@@ -121,16 +121,36 @@ export function getTokenDecimals(
   if (!chainConfig) throw new Error(`chain config for ${chain} not found`);
 
   if (tokenId === 'native') {
-    return chainConfig.nativeTokenDecimals;
+    const { decimals } = getGasToken(chain);
+    return decimals;
   }
 
   const tokenConfig = getTokenById(tokenId);
   if (!tokenConfig) {
     throw new Error('token config not found');
   }
 
-  const decimals = tokenConfig.decimals;
-  return decimals[chainConfig.context] || decimals.default;
+  const { nativeChain, decimals } = tokenConfig;
+
+  const platform = chainToPlatform(chain);
+  const tokenPlatform = chainToPlatform(nativeChain);
+
+  // If the token is native to the chain, return the token's decimals
+  if (platform === tokenPlatform) return decimals;
+
+  // Otherwise, return the minimum of the token's decimals and the platform's max decimals (token bridge)
+  // See: https://github.com/wormhole-foundation/wormhole/blob/main/whitepapers/0003_token_bridge.md#handling-of-token-amounts-and-decimals
+  const maxWrappedDecimals = {
+    Evm: 18,
+    Solana: 8,
+    Sui: 8,
+    Aptos: 8,
+  } as const;
```

Comment:
it be clearer to write this. "if platform is Evm, decimals is 18; otherwise decimals is 8"

did you not do that because eventually it's more complicated when we add Sui/Sei/Cosmos?

Reply:
I'm not sure what the decimals is on Sei, Cosmos, or Near, but it will be a constant, probably 8 or 18. I could look them up

Reply:
It looks like wrapped assets are capped at 8 decimals except on EVM which caps them at 18